OpenRA.Mods.Common/Widgets/Logic/Ingame/SpritePowerMeterLogic.cs

```diff
@@ -0,0 +1,70 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets.Logic.Ingame
+{
+	public class SpritePowerMeterLogic : ChromeLogic
+	{
+		readonly PowerManager powerManager;
+		readonly SpritePowerMeterWidget powerMeter;
+
+		[ObjectCreator.UseCtor]
+		public SpritePowerMeterLogic(World world, Widget widget)
+		{
+			powerManager = world.LocalPlayer.PlayerActor.Trait<PowerManager>();
+			powerMeter = widget.Get<SpritePowerMeterWidget>("POWERMETER");
+		}
+
+		void CheckFlash()
+		{
+			var startWarningFlash = powerManager.PowerState != PowerState.Normal;
+
+			if (powerMeter.LastTotalPowerDisplay != powerMeter.TotalPowerDisplay)
+			{
+				startWarningFlash = true;
+				powerMeter.LastTotalPowerDisplay = powerMeter.TotalPowerDisplay;
+			}
+
+			if (startWarningFlash && powerMeter.WarningFlash <= 0)
+				powerMeter.WarningFlash = powerMeter.WarningFlashDuration;
+		}
+
+		public override void Tick()
+		{
+			powerMeter.LowPower = powerManager.PowerState == PowerState.Low;
+
+			// Number of power units represent each bar
+			var stepSize = powerMeter.PowerUnitsPerBar;
```

Comment:
The comment wording doesn't make sense, and IMO the variable name is self explanatory.
This should be used directly in the calculations below instead of extracting a redundant variable